### PR TITLE
Fix sort command by introducing Contact Index

### PR DESCRIPTION
**Pull request by russelltankaimin**
## TODOs (This PR is ready for review)
- [x] Find out what sort actually does
- [x] Give index attribute to people
- [x] Add index into eduMate.json
```json
{
    "name" : "Bart Lee",
    "phone" : "93288896",
    "email" : "bartlee@gmail.com",
    "address" : "90 Bartley Road",
    "telegramHandle" : "@bartlee",
    "index" : 57,
    "groups" : [ "CCA", "Bartley" ],
    "modules" : [ "BT3102", "BT1101", "CS3240", "CE2134", "CE2112", "CE1102" ]
  },
```
- [x] JsonAdaptedPerson update all their information
- [ ] Add relevant test cases for JsonAdaptedPerson and relevant stuff -- **deferred to another PR**
- [x] editCommand will now be based on ContactIndex, not Index
- [x] viewCommand will be based on ContactIndex, not Index
- [X] sort with reset index capability
Available via `sort index`
- [x] Index Assigner when you add a command
- [x] Handle contact index not found!
- [x] Add Command with ContactIndex
- [x] Delete Command with ContactIndex
- [ ] User Guide (so teammates will be notified of changes) -- **deferred**

### CODE REVIEW

**Comment by mynameizzhafeez** [comment]
[Sort Command Implementation PR](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/76)

**Comment by russelltankaimin** [comment]
Testing will be deferred to a separate PR. Please note!

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 1]
```diff
@@ -47,7 +47,15 @@ public class SortCommandParser implements Parser<SortCommand> {
     @Override
     public SortCommand parse(String args) throws ParseException {
         requireNonNull(args);
-
+        if (args.trim().equals("index")) {
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:50]: I’m a bit unsure on having users type “index” cos i feel that sort on its own should just sort by index.

Comment (mynameizzhafeez) [reply src/main/java/seedu/address/logic/parser/SortCommandParser.java:50]: Because we want to make the index the ultimate tie breaker.

**Review by mynameizzhafeez — APPROVED** [review]
LGTM. Just may need to discuss the sort index part.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -162,7 +164,8 @@ protected static User createEditedUser(User userToEdit, EditPersonDescriptor edi
                 .orElse(userToEdit.getImmutableModuleTags());
 
         return new User(updatedName, updatedPhone, updatedEmail,
-                updatedAddress, updatedTelegramHandle, updatedGroupTags, updatedModuleTags);
+                updatedAddress, updatedTelegramHandle, new ContactIndex(0),
```
[the remark this answers is not in the document]
Comment (russelltankaimin) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:167]: Users will always have index 0 so that contacts can start from 1 onwards!

**Pull request merged**